File: java/src/main/java/_27/RemoveElement.c
int removeElement(int* nums, int numsSize, int val) {
    if (numsSize == 0)
        return numsSize;
    int i = -1;
    int j = i + 1;
    while (j < numsSize)
    {
        if (nums[j] != val)
        {
            nums[++i] = nums[j];
        }
        j++;
    }
    return i + 1;
}

File: java/src/main/java/_27/RemoveElement.java
package _27;

public class RemoveElement {
    public int removeElement(int[] nums, int val) {
        int endIndex = 0;
        for (int walk = endIndex; walk < nums.length; ++ walk) {
            if (nums[walk] != val) {
                nums[endIndex ++] = nums[walk];
            }
        }
        return endIndex;
    }
}
